crypto/rsa/rsa_sp800_56b_gen.c

```diff
@@ -209,6 +209,143 @@ static int rsa_validate_rng_strength(EVP_RAND_CTX *rng, int nbits)
     return 1;
 }
 
+
+DEFINE_STACK_OF(BIGNUM)
+
+/*
+ *
+ * Using p & q, calculate other required parameters such as n, d.
+ * as well as the CRT parameters dP, dQ, qInv.
+ *
+ * See SP800-56Br1
+ *   6.3.1.1 rsakpg1 - basic (Steps 3-4)
+ *   6.3.1.3 rsakpg1 - crt   (Step 5)
+ *
+ * Params:
+ *     rsa An rsa object.
+ *     nbits The key size.
+ *     e The public exponent.
+ *     ctx A BN_CTX object.
+ * Notes:
+ *   There is a small chance that the generated d will be too small.
+ * Returns: -1 = error,
+ *           0 = d is too small,
+ *           1 = success.
+ *
+ * Note this is a partial duplication of ossl_rsa_sp800_56b_derive_params_from_pq, but returns
+ * our exponents and coefficients as parameters (exps and coeffs), meant for use when importing 
+ * data to an rsa key with ossl_rsa_fromdata, prior to the rsa key having all its other parameters
+ * set properly
+ */
+int ossl_rsa_sp800_56b_build_exp_coeffs_from_pq(BIGNUM *p, BIGNUM *q, BIGNUM *d, BIGNUM *n, int nbits,
+                                             const BIGNUM *e, STACK_OF(BIGNUM) *exps, STACK_OF(BIGNUM) *coeffs, BN_CTX *ctx)
+{
+    int ret = 0;
+    BIGNUM *p1, *q1, *dd = NULL, *dn = NULL, *dp = NULL, *de = NULL, *dq = NULL, *lcm, *p1q1, *gcd;
+    BIGNUM *dmp1, *dmq1, *iqmp;
```

Comment:
You will need to run valgrind on this..
Multiple issues here.. The values such as dmp1 are not freed. And when the free call is added then they need to be initied to NULL..